sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -135,9 +160,129 @@ describe(
+
+      // Modify the data in mock file to have the correct values for
+      // disbursement and application number.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return (
+            fileContent
+              // Set the disbursement number to expected disbursement in multiple detail records.
+              .replace("DISBNUMB01", disbursementId)
+              .replace("DISBNUMB02", disbursementId)
+              .replace("DISBNUMB03", disbursementId)
+              // Set the application number to expected application in multiple detail records.
+              .replace("APPLNUMB01", applicationNumber)
+              .replace("APPLNUMB02", applicationNumber)
+              .replace("APPLNUMB03", applicationNumber)
+              // Set the enrolment confirmation to expected date in multiple detail records.
+              .replace("ENRLDT01", currentDate)
+              .replace("ENRLDT02", currentDate)
+              .replace("ENRLDT03", currentDate)
+          );
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `Disbursement ${disbursement.id}, enrolment confirmed.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+        "Notification has been created to send email to integration contacts.",
+        "Total file parsing errors: 0",
+        "Total detail records found: 4",
+        "Total disbursements found: 2",
+        "Disbursements successfully updated: 1",
+        "Disbursements skipped to be processed: 1",
+        "Disbursements considered duplicate and skipped: 0",
+        "Disbursements failed to process: 0",
+      ];
+      expectedResult.warnings = [
+        "Disbursement 1119353191, record skipped due to reason: Enrolment not found.",
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Expect the notifications to be created.
+      const notifications = await getUnsentECEResponseNotifications(db);
+      expect(notifications).toHaveLength(
+        locationMULT.integrationContacts.length,
+      );
+      const updatedDisbursement = await db.disbursementSchedule.findOne({
+        select: {
+          coeStatus: true,
+          tuitionRemittanceRequestedAmount: true,
+        },
+        where: { id: disbursement.id },
+      });
+      // Expect the COE status of the updated disbursement to be completed.
+      expect(updatedDisbursement.coeStatus).toBe(COEStatus.completed);
+      // The file has 3 detail records for the disbursement, with pay to school amount of 10 in each record.
```

Comment:
Checking tickets #1911 and #1501 I was not able to find the specification about the SUM only for the Y records. If it is already captured there please disregard this comment, otherwise please update the ticket accordingly.

Reply:
I am yet to update the ticket, but thanks for checking and reminding.